### DISCUSSION TITLE
Export types defined in the examples folder

### DISCUSSION THREAD

**Discussion by jasperverbeet** [wagmi-dev/abitype#51] [Ideas]
The [types in the example folder](https://github.com/wagmi-dev/abitype/blob/main/src/examples/types.ts) are not part of the [package export](https://github.com/wagmi-dev/abitype/blob/main/src/index.ts) but offer beneficial types for interacting with smart contracts in a non-wagmi environment.

Currently, I'm working on a project in a node environment where I'm switching from [`Typechain`](https://github.com/dethcrypto/TypeChain) to `abitype` with some custom implementations. To obtain the types for events and functions, I'm relying on the types defined in the examples folder. I have to copy and define them manually, but I'd rather import these types directly from the package.

Is this something that would be considered?

**Comment by jasperverbeet** [comment]
I noticed a lot of these types are exported by `@wagmi/core`, would that be the way to go?

**Reply by tmm** [reply]
What example types are you using? Asking because it could make sense to export these as utility types as well.

**Reply by jasperverbeet** [reply]
Thanks for your response!

I'm working on a `MockProvider` for  Ethers (https://github.com/Byont-Ventures/mocketh). I'd like to mock all native ethereum calls. The first mock: `mockContractCall` is working and  I'm successfully getting the correct return types using `GetReturnType` (copied from here: https://github.com/wagmi-dev/abitype/blob/main/src/examples/types.ts).

Currently, I'm working on mocking events, where I would like to give the user the ability to provide a custom event "result". I got a prototype working for that with "unknown" types. However, while I'm writing this and testing some stuff, I noticed I often could get away with the types that are exported by `abitype` already.

For example, this is my weakened typed signature for mocking events:

```ts
mockEvent<
    TAbi extends Abi,
    TEventName extends ExtractAbiEventNames<TAbi>,
    TAbiEvent extends ExtractAbiEvent<TAbi, TEventName>
  >({
    abi,
    eventName,
    returnValue,
  }: {
    abi: TAbi
    eventName: TEventName
    returnValue: AbiParametersToPrimitiveTypes<TAbiEvent['inputs']>[]
  })
```

In conclusion: exporting the types from the `types.ts` in examples would be useful, so I don't have to declare these myself, but not strictly necessary.

Thanks you!
